gloo-edge/gloo-edge/README.md

```diff
@@ -286,40 +628,52 @@ The browser will warn you that your connection is not private due to the self-si
+And **httpbin**:
 
-Let's start by installing Keycloak:
+> Since you are using self-signed certificates, you need to allow insecure server connection when using SSL with *-k*
 
-```bash
-kubectl create -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/12.0.4/kubernetes-examples/keycloak.yaml
-kubectl rollout status deploy/keycloak
 ```
-**Troubleshooting Tip**
+curl -k $APP_URL/not-secured/get
+```
+### Authentication with OIDC (OpenID Connect)
+
+In many use cases, you need to restrict the access to your applications to authenticated users. 
+
+OIDC (OpenID Connect) is an identity layer on top of the OAuth 2.0 protocol. In OAuth 2.0 flows, authentication is performed by an external Identity Provider (IdP) which, in case of success, returns an Access Token representing the user identity. The protocol does not define the contents and structure of the Access Token, which greatly reduces the portability of OAuth 2.0 implementations.
 
-Make sure that the KeyCloak deployment worked correctly by executing the following.
+The goal of OIDC is to address this ambiguity by additionally requiring Identity Providers to return a well-defined ID Token. OIDC ID tokens follow the JSON Web Token standard and contain specific fields that your applications can expect and handle. This standardization allows you to switch between Identity Providers – or support multiple ones at the same time – with minimal, if any, changes to your downstream services; it also allows you to consistently apply additional security measures like Role-based Access Control (RBAC) based on the identity of your users, i.e. the contents of their ID token.
 
-`kubectl describe pod keycloak`
 
-If you see an issue with the readiness probe failing, then you will want to edit the deployment and add the following values under the readinessProbe.
+In this step, you will secure the **bookinfo** application using an OIDC Identity Provider.
 
-`kubectl patch deployment keycloak --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/readinessProbe/timeoutSeconds", "value":10}]'`
+Let's start by installing Keycloak:
 
-`kubectl patch deployment keycloak --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/readinessProbe/initialDelaySeconds", "value":10 }]'`
+```bash
+kubectl create -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/12.0.4/kubernetes-examples/keycloak.yaml
+kubectl rollout status deploy/keycloak
+```
 
 <!--bash
 sleep 30
 -->
 
-Then, we need to configure it and create two users:
+Then, you need to configure it and create two users:
 
 - User1 credentials: `user1/password`
   Email: user1@solo.io
 
 - User2 credentials: `user2/password`
   Email: user2@example.com
 
+> **Note:** One of the following commands retrieves a token which expires quite soon. If you get a *Not Authorized* error, please, re-run this command:
```

Comment:
The note should be added below the following code block, otherwise many people will try to run the block the note contains. Also, the code block of this note should be part of the note (starting with >)